### ISSUE TITLE
Add support for Gateway API

### ISSUE THREAD

**Issue by somaritane** [k8gb-io/k8gb#954]

**Comment by illrill** [comment]
Eager about this one before we can start considering K8GB for our company 👍

**Comment by abaguas** [comment]
In the past few months we paved the way to enable this integration. I can't promise dates, but we may release this feature around March

**Comment by larssb** [comment]
@abaguas that would be so wonderfully awesome. As we've already transitioned many of our workloads being exposed outside X Kubernetes cluster to be using the Gateway API.

**Comment by angelbarrera92** [comment]
+1 looking for it!

Meanwhile im trying to understand if I can use `service` instead of `ingress` in the `Gslb.spec.resourceRef` attribute. That would help transitioning from ingress to gateway api in diferent clusters.

Service Example: https://github.com/k8gb-io/k8gb/blob/master/docs/examples/lbservice-gslb-example.yaml#L32

TL;DR

cluster-a having ingress for the same workload
cluster-b having gateway-api for the same workload

**Comment by abaguas** [comment]
> +1 looking for it!
> 
> Meanwhile im trying to understand if I can use `service` instead of `ingress` in the `Gslb.spec.resourceRef` attribute. That would help transitioning from ingress to gateway api in diferent clusters.
> 
> Service Example: https://github.com/k8gb-io/k8gb/blob/master/docs/examples/lbservice-gslb-example.yaml#L32
> 
> TL;DR
> 
> cluster-a having ingress for the same workload
> cluster-b having gateway-api for the same workload
> 
> 

@angelbarrera92 
I never tested this setup, but I think it will work. We identify a workload by the hostname. So you only need to make sure the GSLBs on the different clusters refer to the same hostname, independently of how the app is exposed (istio, gateway api, ingress, lb service).

**Comment by angelbarrera92** [comment]
I'll test it in the upcoming days :) (hopefully)

**Comment by ytsarev** [comment]
Gateway support was released in https://github.com/k8gb-io/k8gb/releases/tag/v0.17.0
